### DISCUSSION TITLE
How to deal with directory symlinks? (/mnt access in spotify)

### DISCUSSION THREAD

**Discussion by realnc** [netblue30/firejail#5422] [Q&A] [answered]
I'm trying to run Spotify in firejail 0.9.70. `~/.cache/spotify` is a symlink to `/mnt/Data/cache/spotify`. As a result, spotify will not start in firejail. The window pops up, but freezes with nothing displayed and doesn't react to anything. After a few seconds, it exits.

I've added this to my `spotify.local`, but it doesn't help:

```none
noblacklist /mnt/Data/cache/spotify
whitelist /mnt/Data/cache/spotify
whitelist ${HOME}/.cache/spotify
```

The output is:

```none
Reading profile /etc/firejail/spotify.profile
Reading profile /home/realnc/.config/firejail/spotify.local
Reading profile /etc/firejail/disable-common.inc
Reading profile /etc/firejail/disable-devel.inc
Reading profile /etc/firejail/disable-exec.inc
Reading profile /etc/firejail/disable-interpreters.inc
Reading profile /etc/firejail/disable-programs.inc
Reading profile /etc/firejail/whitelist-common.inc
Reading profile /etc/firejail/whitelist-var-common.inc
Parent pid 9203, child pid 9204
Private /opt installed in 192.92 ms
Warning: skipping none for private /srv
Private /srv installed in 0.03 ms
17 programs installed in 8.94 ms
Warning: skipping alternatives for private /etc
Warning: skipping crypto-policies for private /etc
Warning: skipping pki for private /etc
Private /etc installed in 9.20 ms
Private /usr/etc installed in 0.00 ms
Warning: NVIDIA card detected, nogroups command ignored
Warning: NVIDIA card detected, nogroups command ignored
Warning: not remounting /run/user/1000/doc
Warning: NVIDIA card detected, nogroups command ignored
Warning: cannot create spotify directory
Warning: NVIDIA card detected, nogroups command ignored
Blacklist violations are logged to syslog
Warning: NVIDIA card detected, nogroups command ignored
Warning: NVIDIA card detected, nogroups command ignored
Child process initialized in 277.74 ms
[1014/145128.826692:ERROR:context.cc(151)] The cache_path directory (/home/realnc/.cache/spotify/Browser) could not be created.
[1014/145128.826726:ERROR:context.cc(166)] The cache_path is invalid. Defaulting to in-memory storage.
terminating with uncaught exception of type boost::filesystem::filesystem_error: boost::filesystem::status: Permission denied [system:13]: "/home/realnc/.cache/spotify/Users/realnc-user/offline_state"

Parent is shutting down, bye...
```

It only works when `~/.cache/spotify` is not a symlink. How can I make it work?

**Comment by rusty-snake** [comment, marked as answer]
1. Add `ignore disable-mnt` to allow access to `/mnt`.
2. `/mnt/Data/cache/spotify` is never blacklisted (unless you did), you can remove the `noblacklist`.
3. `whitelist ${HOME}/.cache/spotify` is already in the profile, you can remove it.
4. `whitelist /mnt/Data/cache/spotify` if you have anthing else in `/mnt` that this required by spotify (e.g. `/mnt/Data/config/spority`), it will be invisible to spotify.

**Reply by realnc** [reply]
Thank you! `ignore disable-mnt` makes it work. Is this a new setting? firejail used to allow access to everything on my `/mnt/Data`, so I always had to manually add `blacklist /mnt/Data/*` to my profiles. Apparently this has changed now?

**Reply by rusty-snake** [reply]
firejail supports `disable-mnt` fro the early beginnings and spotify.profile contains it for 5 years.

**Reply by realnc** [reply]
Ah. I was confused by discord. It has `ignore disable-mnt` in `discord-common.profile` and I hadn't noticed it.
